### PR TITLE
AEA-362 End-to-end testing protocol generator

### PR DESCRIPTION
**Pull request by 5A11**
## Proposed changes

Work in progress!

End-to-end test of the protocol generator

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/data/generator/two_party_negotiation/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2020 fetchai Limited
```
Comment (DavidMinarsch) [inline tests/data/generator/two_party_negotiation/__init__.py]: @5A11 please change generator so it only says `Copyright {year} {author}`, i.e. in this case `Copyright 2020 fetchai`

File: tests/data/generator/two_party_negotiation/message.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+            assert (
+                type(self.dialogue_reference[0]) == str
+            ), "The first element of dialogue_reference must be 'str' but it is not."
+            assert (
+                type(self.dialogue_reference[1]) == str
+            ), "The second element of dialogue_reference must be 'str' but it is not."
+            assert type(self.message_id) == int, "message_id is not int"
+            assert type(self.target) == int, "target is not int"
+
+            # Light Protocol 2
```
Comment (DavidMinarsch) [inline tests/data/generator/two_party_negotiation/message.py]: Maybe instead say `Light Protocol Rule 2` so it is more explicit?

Comment (DavidMinarsch) [reply tests/data/generator/two_party_negotiation/message.py]: same below

File: tests/data/generator/two_party_negotiation/__init__.py (hunk 1 above)
Comment (5A11) [reply tests/data/generator/two_party_negotiation/__init__.py]: Done

File: tests/data/generator/two_party_negotiation/message.py (hunk 2 above)
Comment (5A11) [reply tests/data/generator/two_party_negotiation/message.py]: Done

**Pull request merged**